clients/roscpp/src/libros/transport/transport_tcp.cpp

```diff
@@ -276,7 +276,7 @@ bool TransportTCP::connect(const std::string& host, int port)
 
     bool found = false;
     struct addrinfo* it = addr;
-    char namebuf[128];
+    char namebuf[128] = {0};
```

Comment:
This is somewhat misleading. One might assume that `char numebuf [128] = {1}` would initialize all values to 1, but will only initialize the first value with 1 und every other with 0. `char numebuf [128] = {}` might be the better choice as it makes the [zero initialization](https://en.cppreference.com/w/cpp/language/zero_initialization) obvious.

Reply:
you are right. I will update it. Thanks